sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/EmailClientImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.kyriba.conference.sponsorship.service;
+
+import com.kyriba.conference.sponsorship.domain.EmailMessage;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * @author Aliaksandr Samal
+ */
+@Component
+@RequiredArgsConstructor
+public class EmailClientImpl implements EmailClient
+{
+  @Value("${notification.sync}")
+  private boolean isSyncNotification;
+  private final EmailClientSync emailClientSync;
+  private final EmailClientAsync emailClientAsync;
+
+  @Override
+  public void sendNotification(EmailMessage emailMessage)
```

Comment:
You could implement this logic just as conditional bean configuration

Reply:
Done